### PR TITLE
FURNACE-127: fixed NPE when relativePath is empty

### PR DESCRIPTION
**Pull request by mareknovotny**
https://issues.jboss.org/browse/FURNACE-127

### CODE REVIEW

**Comment by gastaldi** [comment]
Add to whitelist

File: manager/resolver/maven/src/main/java/org/jboss/forge/furnace/manager/maven/ClasspathWorkspaceReader.java [hunk 1]
```diff
@@ -418,10 +418,13 @@ private Artifact createFoundArtifact(final File pomFile)
             if (parent != null)
             {
                Xpp3Dom relativePathNode = parent.getChild("relativePath");
-               String relativePath = (relativePathNode == null) ? "../pom.xml" : relativePathNode.getValue();
-               File parentPom = pomFile.getParentFile().toPath().resolve(relativePath).toFile();
-               if (parentPom.isFile())
-                  result = createFoundModules(parentPom);
+               if ( relativePathNode != null && !(relativePathNode.getValue() == null)  ) 
+               {
+	               String relativePath = (relativePathNode == null) ? "../pom.xml" : relativePathNode.getValue();
```
Comment (gastaldi) [inline manager/resolver/maven/src/main/java/org/jboss/forge/furnace/manager/maven/ClasspathWorkspaceReader.java]: relativePathNode will never be null in here

File: manager/resolver/maven/src/main/java/org/jboss/forge/furnace/manager/maven/ClasspathWorkspaceReader.java [hunk 2]
```diff
@@ -418,10 +418,13 @@ private Artifact createFoundArtifact(final File pomFile)
             if (parent != null)
             {
                Xpp3Dom relativePathNode = parent.getChild("relativePath");
-               String relativePath = (relativePathNode == null) ? "../pom.xml" : relativePathNode.getValue();
-               File parentPom = pomFile.getParentFile().toPath().resolve(relativePath).toFile();
-               if (parentPom.isFile())
-                  result = createFoundModules(parentPom);
+               if ( relativePathNode != null && !(relativePathNode.getValue() == null)  )
```
Comment (gastaldi) [inline manager/resolver/maven/src/main/java/org/jboss/forge/furnace/manager/maven/ClasspathWorkspaceReader.java]: This looks better:
    if ( relativePathNode != null && relativePathNode.getValue() != null )

File: manager/resolver/maven/src/main/java/org/jboss/forge/furnace/manager/maven/ClasspathWorkspaceReader.java [hunk 3]
```diff
@@ -418,10 +418,13 @@ private Artifact createFoundArtifact(final File pomFile)
...
             {
                Xpp3Dom relativePathNode = parent.getChild("relativePath");
-               String relativePath = (relativePathNode == null) ? "../pom.xml" : relativePathNode.getValue();
-               File parentPom = pomFile.getParentFile().toPath().resolve(relativePath).toFile();
-               if (parentPom.isFile())
-                  result = createFoundModules(parentPom);
+               if ( relativePathNode != null && !(relativePathNode.getValue() == null)  ) 
+               {
+	               String relativePath = (relativePathNode == null) ? "../pom.xml" : relativePathNode.getValue();
+	               File parentPom = pomFile.getParentFile().toPath().resolve(relativePath).toFile();
```
Comment (mareknovotny) [inline manager/resolver/maven/src/main/java/org/jboss/forge/furnace/manager/maven/ClasspathWorkspaceReader.java]: you're right, it is much cleaner

**Comment by gastaldi** [comment]
Thanks!

**Comment by gastaldi** [comment]
Retest this please

**Pull request merged**